crates/core/src/factories/core_fetch.rs

```diff
@@ -241,6 +244,31 @@ impl Fetch for CoreFetch {
             Ok(())
         })
     }
+
+    fn handle_incoming_ops(
+        &self,
+        ops: Vec<op_store::MetaOp>,
+    ) -> BoxFut<'_, K2Result<()>> {
+        Box::pin(async move {
+            match self.op_store.process_incoming_ops(ops.clone()).await {
```

Comment:
I was wondering if there should be some protection against invalid ops or DoS attacks here. But it should probably be in the module that owns the fetch module.

Reply:
Indeed this is already partially handled by the module message callback, when you throw a decode error in there it'll close the connection. As far as ddos, we'll need to handle that through some future rate-limit module.

Reply:
I've been assuming that the host will deal with that. We have data validation and a pending rate limiting feature with entry-rate-weights. Though that might need some thought to make it useful.